### PR TITLE
Feat: Sign up page functionality

### PR DESCRIPTION
**Pull request by mariusjagminas**

### CODE REVIEW

File: packages/webapp/src/store/session/session.saga.ts [hunk 1]
```diff
@@ -3,20 +3,28 @@ import { History } from 'history'
...
 export const DELAY_IN_MILISECONDS = 10 * 60000
 
-// eslint-disable-next-line require-unicode-regexp
-const regex = new RegExp(/confirm-email[\S\s]*/)
+const shouldOmitValidation = (history: History, routes: string[]): boolean => {
+  const {
+    location: { pathname },
+  } = history
+  // eslint-disable-next-line require-unicode-regexp
+  const regex = new RegExp(/confirm-email[\S\s]*/)
```
Comment (kowalski) [inline packages/webapp/src/store/session/session.saga.ts:19]: Can't we include the confirm-email route in `whitelistedRoutes`? 
You could have `whilelistedRoutes` be a a list of Regexps and do sth like:

```
return Boolean(routes.find((regexp) => regex.test(pathname))
```

File: packages/webapp/src/pages/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    )
+  }
+
+  return (
+    <PageWrapper>
+      <form onSubmit={handleSubmit(onSubmitHandler)}>
+        <Controller
+          control={control}
+          name="email"
+          render={({ onChange }) => <TextInput onChange={onChange} placeholder="Email" />}
```
Comment (kowalski) [inline packages/webapp/src/pages/SignUp/SignUp.tsx:50]: I think you should also be passing `value` and `onBlur` to all `<TextInput>`

File: packages/webapp/src/pages/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      </PageWrapper>
+    )
+  }
+
+  return (
+    <PageWrapper>
+      <form onSubmit={handleSubmit(onSubmitHandler)}>
+        <Controller
+          control={control}
+          name="email"
```
Comment (kowalski) [inline packages/webapp/src/pages/SignUp/SignUp.tsx:49]: Would be nice to validate that this is an email.

Something like
```
          rules={{
            required: 'This field is required',
            pattern: {
              value: emailRegexp,
              message: 'Is not a correct email address',
            },
          }}
```
```
const emailRegexp = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
```

File: packages/webapp/src/pages/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        <Controller
+          control={control}
+          name="password"
+          render={({ onChange }) => <TextInput onChange={onChange} placeholder="Password" />}
+        />
+        <br />
+        <br />
+        <Controller
+          control={control}
+          name="confirmPassword"
```
Comment (kowalski) [inline packages/webapp/src/pages/SignUp/SignUp.tsx:63]: you should validate that confirm matches password.

You can do it like:

```
          rules={{ validate: validateConfirmation }}
```

```
  const validateConfirmation = (value: string) => {
    if (value !== watch('email')) return 'Passwords do not match';
  };

```

(`watch` is part of `useForm` result)

**Review by kowalski — CHANGES_REQUESTED** [review]
Once you add validation for individual fields it's a good idea to display the error messages.

The `useForm` hook gives you `errors` object, that has an error for each field.
So you could extend <TextInput>  to get the optional `error` prop and pass it like:

```
              error={errors.email?.message}

```

File: packages/webapp/src/pages/SignUp/SignUp.tsx (hunk 2 above)
Comment (mariusjagminas) [reply packages/webapp/src/pages/SignUp/SignUp.tsx:50]: Console displays an error "Warning: A component is changing an uncontrolled input of type text to be controlled" when I pass `value` to <TextInput/> .I should figure out how to set initial value as an empty string to overcome this error.

File: packages/webapp/src/pages/SignUp/SignUp.tsx (hunk 4 above)
Comment (mariusjagminas) [reply packages/webapp/src/pages/SignUp/SignUp.tsx:63]: I do that in `onSubmitHandler` function
 ```
 try {
      if (password !== confirmPassword) {
        throw new Error(ErrorMsg.PASSWORDS_DONT_MATCH)
      }
```

File: packages/webapp/src/pages/SignUp/SignUp.tsx (hunk 2 above)
Comment (kowalski) [reply packages/webapp/src/pages/SignUp/SignUp.tsx:50]: a, that's easy, add `defaultValue=''` to Controller

File: packages/webapp/src/pages/SignUp/SignUp.tsx (hunk 4 above)
Comment (kowalski) [reply packages/webapp/src/pages/SignUp/SignUp.tsx:63]: It's better to use react-hook library as it was intendented by their creators.
The validation is done way before submit.
Submit is not even possible if any of the fields have error.

**Review by kowalski — APPROVED** [review]

**Pull request merged**